### ISSUE TITLE
Expected 2D array, got 1D array instead

### ISSUE THREAD

**Issue by jordanlmx** [boudinfl/pke#67]
Hi I face a problem when I try to training the KEA model with files provided in the example
```
Traceback (most recent call last):

  File "<ipython-input-7-322b61673763>", line 1, in <module>
    runfile('C:/Users/123/Desktop/PKE/train/test/Train_model.py', wdir='C:/Users/123/Desktop/PKE/train/test')

  File "C:\Users\123\Anaconda3\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 668, in runfile
    execfile(filename, namespace)

  File "C:\Users\123\Anaconda3\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 108, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "C:/Users/123/Desktop/PKE/train/test/Train_model.py", line 32, in <module>
    model=pke.supervised.Kea())

  File "C:\Users\123\Anaconda3\lib\site-packages\pke\utils.py", line 216, in train_supervised_model
    model_file=model_file)

  File "C:\Users\123\Anaconda3\lib\site-packages\pke\supervised\feature_based\kea.py", line 168, in train
    clf.fit(training_instances, training_classes)

  File "C:\Users\123\Anaconda3\lib\site-packages\sklearn\naive_bayes.py", line 585, in fit
    X, y = check_X_y(X, y, 'csr')

  File "C:\Users\123\Anaconda3\lib\site-packages\sklearn\utils\validation.py", line 749, in check_X_y
    estimator=estimator)

  File "C:\Users\123\Anaconda3\lib\site-packages\sklearn\utils\validation.py", line 549, in check_array
    "if it contains a single sample.".format(array))

ValueError: Expected 2D array, got 1D array instead:
array=[].
Reshape your data either using array.reshape(-1, 1) if your data has a single feature or array.reshape(1, -1) if it contains a single sample.
```

I am not sure what problem is it. 
BTW I face another problem when I try to train the model with my own data (which is in ‘raw’ format)
No error was given but I get following warning:
```
naive_bayes.py:465: RuntimeWarning: divide by zero encountered in log
  self.class_log_prior_ = (np.log(self.class_count_) -
```
And I test with the trained model, the result keyphrases generated are all with zero scores.

Thanks in advance for solving my problem

**Comment by boudinfl** [comment]
Hi @jordanlmx,

I am not sure of what is happening with your code. Can you provide me give me an example of file and your code so I can reproduce this issue on my computer ?

Best,

f.

**Comment by jordanlmx** [comment]
I have send you an email please check Thank you

**Comment by jordanlmx** [comment]
Hi how was going on?
Did you manage to found the problem?

**Comment by boudinfl** [comment]
Hi @jordanlmx,

I found the issues with your code:

1. The code-breaking issue is that you give a bad path to the test file:
```
extractor.load_document(input='test_doc/mcdonalds_train.txt', # <- this file does not exists 
                                           language='en')
```
2. your DF counts file is empty, i.e. it contains no words \tab count lines apart from the _--NB_DOC--	0--. Please have a look at the code documentation [here](https://boudinfl.github.io/pke/build/html/tutorials/df.html) to solve this issue.

f.

**Comment by jordanlmx** [comment]
Hi I think I may send you the wrong file
I have attached new file in latest email.
Please check. Thank you for your help.

**Comment by boudinfl** [comment]
I have no issues running your code on my machine. Here is what I get:

```
pc-finot:train_model florian$ python3 Train_model.py 
INFO:root:Loading df counts from df/DF.gz
INFO:root:building model <pke.supervised.feature_based.kea.Kea object at 0x105a75ef0> from train_doc/
INFO:root:loading reference keyphrases from reference.txt
INFO:root:reading file train_doc/kfc.txt
INFO:root:reading file train_doc/mcdonalds.txt
INFO:root:writing model to model.pickle
pc-finot:train_model florian$ python3 Kea_Test.py 
kfc 0.0255119428229
willing 0.0132026572254
pieces of chicken 0.012886876127
pcs 0.0127646235141
pls 0.0127583287924
kfc delivery 0.0124299428747
birthday treat 0.0123172631288
chizza 0.0123135155005
personal size 0.0123037206958
personal size pizza 0.0123037206958
```

Can you try to update your `pke` version and all its dependencies (sklearn, etc.) ?

f.

**Comment by jordanlmx** [comment]
Oh that's weird
OK I will try to do so
Thank you for your help
I will come back if there is any issues

**Comment by jordanlmx** [comment]
Urghh
I am not sure why I am still facing the same problem
Maybe I can post my version of pke and other dependencies here

```
Libraries         Version
======          =====
six             1.11.0
spacy           2.0.16
nltk            3.3.0
networkx        2.2
numpy           1.15.4
scipy           1.1.0
scikit-learn    0.20.0
unidecode       1.0.23
future          0.17.1
pke             1.8
```
Sorry for disturbing you.

**Comment by jordanlmx** [comment]
Hi @boundinfl may I know about your version for each libraries?
Maybe that is the issues

**Comment by boudinfl** [comment]
The same version except I have 0.19.1 for scikit-learn and 1.14.2 for numpy.

**Comment by jordanlmx** [comment]
Hi the problem still exist. I have already try with same version of libraries
Can you send me your naives_bayes.py in scikit-learn library
because the runtime warning occur in this file, maybe I can check what happened in it

**Comment by vitordouzi** [comment]
I also found this problem when the KEA (or even WINGNUS) doesn't find any candidates to create the features on `feature_extraction` function

To fix this, I change the `train_supervised_model` function on utils.py file adding the following lines
```
if not len(model.candidates):
      continue
```

after `model.candidate_selection()` line.

with this, the code skips all documents without candidates and fixed this problem for my case.

**Comment by boudinfl** [comment]
This is fixed in PR #84.
